lib/ui/semantics.dart

```diff
@@ -641,6 +641,121 @@ class SemanticsFlag {
   }
 }
 
+/// A string that carries a list of [StringAttribute]s.
+class AttributedString extends NativeFieldWrapperClass2 {
+  /// Creates a attributed string.
+  ///
+  /// The [TextRange] in the [attributes] must be inside the length of the
+  /// [string].
+  AttributedString(
+    String string, {
+    List<StringAttribute>? attributes,
+  }) : assert((){
+         if (attributes != null) {
+           for (final StringAttribute attribute in attributes) {
+             assert(
+               string.length >= attribute.range.start &&
+               string.length >= attribute.range.end,
+               'The range in $attribute is outside of the string $string',
+             );
+           }
+         }
+         return true;
+       }()) {
+    _constructor();
+    _initAttributedString(string, attributes);
+  }
+
+  AttributedString._withConcat(
+    AttributedString first,
+    AttributedString second
+  ) {
+    _constructor();
+    _initWithConcat(first, second);
+  }
+
+  /// The plain string stored in the attributed string.
+  String get string => _getString();
+
+  void _constructor() native 'AttributedString_constructor';
+  void _initAttributedString(String string, List<StringAttribute>? attributes) native 'AttributedString_initAttributedString';
+  void _initWithConcat(AttributedString first, AttributedString second) native 'AttributedString_initWithConcat';
+  String _getString() native 'AttributedString_GetString';
+  String _toString() native 'AttributedString_toString';
+
+  @override
+  AttributedString operator +(AttributedString other) {
+    return AttributedString._withConcat(this, other);
+  }
+
+  @override
+  String toString() {
+    return _toString();
+  }
+}
+
+// When adding a new StringAttribute, the classes in these files must be
+// updated as well.
+//  * engine/src/flutter/lib/web_ui/lib/src/ui/semantics.dart
+//  * engine/src/flutter/lib/ui/semantics/string_attribute.h
+//  * engine/src/flutter/shell/platform/android/io/flutter/view/AccessibilityBridge.java
+
+/// An abstract interface for string attribute that affects how assistive
+/// technologies, e.g. VoiceOver or TalkBack, treat the text.
+///
+/// See also:
+///
+///  * [SpellOutStringAttribute], which causes the assistive technologies to
+///    spell out the string character by character when announcing the string.
+///  * [LocaleStringAttribute], which causes the assistive technologies to
+///    treat the string in the specific language.
+abstract class StringAttribute extends NativeFieldWrapperClass2 {
```

Comment:
This class doesn't need to be `NativeFieldWrapper2`. We should avoid sub classes needing to make multiple hops to native to construct/initialize themselves. We can just pass `this` into the native side and use `AssosicateWithDartWrapper`.

Reply:
I tried using
```dart
abstract class StringAttribute {}

class LocaleStringAttribute extends StringAttribute with NativeFieldWrapperClass2{}

class SpellOutStringAttribute extends StringAttribute with NativeFieldWrapperClass2{}
```
but it throw weird error when i create the String attributes
```
[ERROR:flutter/shell/common/shell.cc(93)] Dart Error: Dart_SetNativeInstanceField: invalid index 0 passed in to set native instance field
[ERROR:flutter/shell/common/shell.cc(93)] assertion failed !LogIfError(Dart_SetNativeInstanceField( wrapper, kPeerIndex, reinterpret_cast<intptr_t>(this)))
[ERROR:flutter/fml/backtrace.cc(118)] Caught signal SIGABRT during program execution.
Frame 0: 0x7fff2030d720 abort
Frame 1: 0x101ebfa41 tonic::DartWrappable::AssociateWithDartWrapper(_Dart_Handle*)
```

Reply:
I guess the NativeFieldWrapperClass2 can't be use as mixin?

Reply:
I think you can just make StringAttribute extend NativeFieldWrapper2 and then make the others extend it and it'll work. I don't think mixing in will work.

Reply:
That is what i was using, I will resolve this comments